nbhttp/engine.go

```diff
@@ -698,7 +702,7 @@ func NewEngineTLS(conf Config, v ...interface{}) *Engine {
 		engine.conns[c] = struct{}{}
 		engine.mux.Unlock()
 		engine._onOpen(c)
-		tlsConn := tls.NewConn(c, engine.tlsConfig, isClient, true, mempool.DefaultMemPool)
+		tlsConn := tls.NewConn(c, engine.tlsConfig, isClient, true, conf.TLSAllocator)
```

Comment:
We should use the same allocator for TLS and nbio.Conn, then they can reuse the buffers together.

Reply:
I did this so I could figure out if the tls connection code is leaking

Reply:
ok, merged, let's see it.